### PR TITLE
feat(selling): added sales target to customer

### PR DESCRIPTION
**Pull request by MyuddinKhatri**
added sales target to the customer and displayed Monthly Sales Target and Total Monthly Sales as progress in Customer Dashboard

Sales Target Fields
sales_target

Sales Target Dashboard
sales_target_dashboard

### CODE REVIEW

File: erpnext/patches/v13_0/update_customer_total_sales.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import frappe
+from erpnext.selling.doctype.customer.customer import update_customer_current_month_sales
+
+def execute():
+	'''Update customer monthly sales history based on sales invoices'''
+	frappe.reload_doctype("Customer")
+	customers = [d['name'] for d in frappe.get_list("Customer")]
+
+	for customer in customers:
+		update_customer_current_month_sales(customer)
```
Comment (rohitwaghchaure) [inline erpnext/patches/v13_0/update_customer_total_sales.py:14]: This patch will take lot of time if the number of customers are more than 1,00,000

File: erpnext/patches.txt [hunk 2]
```diff
@@ -707,3 +707,6 @@ erpnext.patches.v13_0.move_doctype_reports_and_notification_from_hr_to_payroll #
 erpnext.patches.v13_0.move_payroll_setting_separately_from_hr_settings #22-06-2020
 erpnext.patches.v13_0.check_is_income_tax_component #22-06-2020
 erpnext.patches.v12_0.add_taxjar_integration_field
+erpnext.patches.v12_0.update_sales_taxes_and_charges_template
```
Comment (rohitwaghchaure) [inline erpnext/patches.txt]: What is the use of this patch and why the file of the patch not showing in this PR

File: erpnext/patches.txt [hunk 3]
```diff
@@ -707,3 +707,6 @@ erpnext.patches.v13_0.move_doctype_reports_and_notification_from_hr_to_payroll #
...
+erpnext.patches.v13_0.update_customer_total_sales
+erpnext.patches.v13_0.setup_remove_assignment_on_task
```
Comment (rohitwaghchaure) [inline erpnext/patches.txt]: What is the use of this patch and why the file of the patch not showing in this PR

File: erpnext/patches/v13_0/update_customer_total_sales.py (hunk 1 above)
Comment (MyuddinKhatri) [reply erpnext/patches/v13_0/update_customer_total_sales.py:14]: @rohitwaghchaure  Can you please suggest something for this?

Comment (rohitwaghchaure) [reply erpnext/patches/v13_0/update_customer_total_sales.py:14]: get customer wise sales data from sales invoice using sql query(group by customer) and then update the value in the customer. If particular instance has 10,000 customer out of which only 100 customers has the sales invoices and not all then no need to check for each customer.

**Pull request closed without merging**